File: jp.go.aist.rtm.rtcbuilder.vbdotnet/resource/ConfigSet/configset4/foo.vb
Imports RTC
Imports omg.org.RTC



Public Class foo
    Inherits DataFlowComponentBase



    Private m_int_param0Conf As int = New int()
    Private Function int_param0Trans(ByVal str As String) As Boolean
                Return int.TryParse(str, m_int_param0Conf)
    End Function
    Private m_float_param0Conf As List(Of float) = New List(Of float)()
    Private Function float_param0Trans(ByVal str As String) As Boolean
                Dim items As String = str.Split(New Char() {","})
                m_float_param0Conf.Clear()
                Dim ret As Boolean = True
                For Each item As String in items
                    Dim temp As float
                    if !float.TryParse(item, temp) Then
                        ret = True
                        Continue For
                    End If
                    m_float_param0Conf.Add(temp)
                Next
                Return ret
    End Function


    Public Sub New(ByVal manager As Manager)
        MyBase.New(manager)


    End Sub

    Protected Overrides Function onInitialize() As ReturnCode_t
        bindParameter("int_param0", "0", New RTC.Config.TransFunc(AddressOf int_param0Trans))
        bindParameter("float_param0", "1.0,2.0,3.0", New RTC.Config.TransFunc(AddressOf float_param0Trans))

        Return ReturnCode_t.RTC_OK
    End Function

    'Protected Overrides Function onFinalize() As ReturnCode_t
    '    Return ReturnCode_t.RTC_OK
    'End Function

    'Protected Overrides Function onStartup(ByVal ec_id As Integer) As ReturnCode_t
    '    Return ReturnCode_t.RTC_OK
    'End Function

    'Protected Overrides Function onShutdown(ByVal ec_id As Integer) As ReturnCode_t
    '    Return ReturnCode_t.RTC_OK
    'End Function

    'Protected Overrides Function onActivated(ByVal ec_id As Integer) As ReturnCode_t
    '    Return ReturnCode_t.RTC_OK
    'End Function

    'Protected Overrides Function onDeactivated(ByVal ec_id As Integer) As ReturnCode_t
    '    Return ReturnCode_t.RTC_OK
    'End Function

    'Protected Overrides Function onExecute(ByVal ec_id As Integer) As ReturnCode_t
    '    Return ReturnCode_t.RTC_OK
    'End Function

    'Protected Overrides Function onAborting(ByVal ec_id As Integer) As ReturnCode_t
    '    Return ReturnCode_t.RTC_OK
    'End Function

    'Protected Overrides Function onError(ByVal ec_id As Integer) As ReturnCode_t
    '    Return ReturnCode_t.RTC_OK
    'End Function

    'Protected Overrides Function onReset(ByVal ec_id As Integer) As ReturnCode_t
    '    Return ReturnCode_t.RTC_OK
    'End Function

    'Protected Overrides Function onStateUpdate(ByVal ec_id As Integer) As ReturnCode_t
    '    Return ReturnCode_t.RTC_OK
    'End Function

    'Protected Overrides Function onRateChanged(ByVal ec_id As Integer) As ReturnCode_t
    '    Return ReturnCode_t.RTC_OK
    'End Function
End Class

Public Class fooInit

    Private spec() As String = New String() _
    { _
        "implementation_id", "foo", _
        "type_name",         "foo", _
        "description",       "MDesc", _
        "version",           "1.0.3", _
        "vendor",            "TA2", _
        "category",          "manip2", _
        "activity_type",     "DataFlowComponent", _
        "max_instance",      "3", _
        "language",          "Visual Basic", _
        "lang_type",         "COMPILE" _
        , "conf.default.int_param0",   "0" _
        , "conf.default.float_param0",   "1.0,2.0,3.0" _
    }

    Public Sub New(ByVal manager As Manager)
        Dim profile As Properties = New Properties(spec)
        manager.registerFactory(profile, _
            New RtcNewFunc(AddressOf Createfoo), _
            New RtcDeleteFunc(AddressOf Deletefoo))
    End Sub

    Public Function Createfoo(ByVal manager As Manager) As RTObject_impl
        Return New foo(manager)
    End Function

    Public Sub Deletefoo(ByVal rtc As RTObject_impl)
        rtc.Dispose()
    End Sub

End Class

File: jp.go.aist.rtm.rtcbuilder.python/resource/100/ConfigSet/ConfigSetType/foo.py
#!/usr/bin/env python
# -*- coding: utf-8 -*-
# -*- Python -*-
# <rtc-template block="description">
"""
 @file foo.py
 @brief MDesc
 @date $Date$
"""
# </rtc-template>
import sys
import time
sys.path.append(".")
# Import RTM module
import RTC
import OpenRTM_aist
# Import Service implementation class
# <rtc-template block="service_impl">
# </rtc-template>
# Import Service stub modules
# <rtc-template block="consumer_import">
# </rtc-template>
# This module's spesification
# <rtc-template block="module_spec">
foo_spec = ["implementation_id", "foo", 
         "type_name",         "foo", 
         "description",       "MDesc", 
         "version",           "1.0.3", 
         "vendor",            "TA2", 
         "category",          "manip2", 
         "activity_type",     "STATIC2", 
         "max_instance",      "3", 
         "language",          "Python", 
         "lang_type",         "SCRIPT",
         "conf.default.short_param", "0",
         "conf.default.int_param", "1",
         "conf.default.long_param", "14",
         "conf.default.float_param", "0.11",
         "conf.default.double_param", "4.11",
         "conf.default.str_param0", "hoge",
         "conf.default.str_param1", "dara",
         "conf.__type__.short_param", "short",
         "conf.__type__.int_param", "int",
         "conf.__type__.long_param", "long",
         "conf.__type__.float_param", "float",
         "conf.__type__.double_param", "double",
         "conf.__type__.str_param0", "string",
         "conf.__type__.str_param1", "string",
         ""]
# </rtc-template>
# <rtc-template block="component_description">
##
# @class foo
# @brief MDesc
# 
# 
# </rtc-template>
class foo(OpenRTM_aist.DataFlowComponentBase):
	
    ##
    # @brief constructor
    # @param manager Maneger Object
    # 
    def __init__(self, manager):
        OpenRTM_aist.DataFlowComponentBase.__init__(self, manager)
		
        # initialize of configuration-data.
        # <rtc-template block="init_conf_param">
        """
        
         - Name:  short_param
         - DefaultValue: 0
        """
        self._short_param = [0]
        """
        
         - Name:  int_param
         - DefaultValue: 1
        """
        self._int_param = [1]
        """
        
         - Name:  long_param
         - DefaultValue: 14
        """
        self._long_param = [14]
        """
        
         - Name:  float_param
         - DefaultValue: 0.11
        """
        self._float_param = [0.11]
        """
        
         - Name:  double_param
         - DefaultValue: 4.11
        """
        self._double_param = [4.11]
        """
        
         - Name:  str_param0
         - DefaultValue: hoge
        """
        self._str_param0 = ['hoge']
        """
        
         - Name:  str_param1
         - DefaultValue: dara
        """
        self._str_param1 = ['dara']
		
        # </rtc-template>
		 
    ##
    #
    # The initialize action (on CREATED->ALIVE transition)
    # 
    # @return RTC::ReturnCode_t
    # 
    #
    def onInitialize(self):
        # Bind variables and configuration variable
        self.bindParameter("short_param", self._short_param, "0")
        self.bindParameter("int_param", self._int_param, "1")
        self.bindParameter("long_param", self._long_param, "14")
        self.bindParameter("float_param", self._float_param, "0.11")
        self.bindParameter("double_param", self._double_param, "4.11")
        self.bindParameter("str_param0", self._str_param0, "hoge")
        self.bindParameter("str_param1", self._str_param1, "dara")
		
        # Set InPort buffers
		
        # Set OutPort buffers
		
        # Set service provider to Ports
		
        # Set service consumers to Ports
		
        # Set CORBA Service Ports
		
        return RTC.RTC_OK
	
    ###
    ## 
    ## The finalize action (on ALIVE->END transition)
    ## 
    ## @return RTC::ReturnCode_t
    #
    ## 
    #def onFinalize(self):
    #
    #    return RTC.RTC_OK
	
    ###
    ##
    ## The startup action when ExecutionContext startup
    ## 
    ## @param ec_id target ExecutionContext Id
    ##
    ## @return RTC::ReturnCode_t
    ##
    ##
    #def onStartup(self, ec_id):
    #
    #    return RTC.RTC_OK
	
    ###
    ##
    ## The shutdown action when ExecutionContext stop
    ##
    ## @param ec_id target ExecutionContext Id
    ##
    ## @return RTC::ReturnCode_t
    ##
    ##
    #def onShutdown(self, ec_id):
    #
    #    return RTC.RTC_OK
	
    ###
    ##
    ## The activated action (Active state entry action)
    ##
    ## @param ec_id target ExecutionContext Id
    ## 
    ## @return RTC::ReturnCode_t
    ##
    ##
    #def onActivated(self, ec_id):
    #
    #    return RTC.RTC_OK
	
    ###
    ##
    ## The deactivated action (Active state exit action)
    ##
    ## @param ec_id target ExecutionContext Id
    ##
    ## @return RTC::ReturnCode_t
    ##
    ##
    #def onDeactivated(self, ec_id):
    #
    #    return RTC.RTC_OK
	
    ###
    ##
    ## The execution action that is invoked periodically
    ##
    ## @param ec_id target ExecutionContext Id
    ##
    ## @return RTC::ReturnCode_t
    ##
    ##
    #def onExecute(self, ec_id):
    #
    #    return RTC.RTC_OK
	
    ###
    ##
    ## The aborting action when main logic error occurred.
    ##
    ## @param ec_id target ExecutionContext Id
    ##
    ## @return RTC::ReturnCode_t
    ##
    ##
    #def onAborting(self, ec_id):
    #
    #    return RTC.RTC_OK
	
    ###
    ##
    ## The error action in ERROR state
    ##
    ## @param ec_id target ExecutionContext Id
    ##
    ## @return RTC::ReturnCode_t
    ##
    ##
    #def onError(self, ec_id):
    #
    #    return RTC.RTC_OK
	
    ###
    ##
    ## The reset action that is invoked resetting
    ##
    ## @param ec_id target ExecutionContext Id
    ##
    ## @return RTC::ReturnCode_t
    ##
    ##
    #def onReset(self, ec_id):
    #
    #    return RTC.RTC_OK
	
    ###
    ##
    ## The state update action that is invoked after onExecute() action
    ##
    ## @param ec_id target ExecutionContext Id
    ##
    ## @return RTC::ReturnCode_t
    ##
    ##
    #def onStateUpdate(self, ec_id):
    #
    #    return RTC.RTC_OK
	
    ###
    ##
    ## The action that is invoked when execution context's rate is changed
    ##
    ## @param ec_id target ExecutionContext Id
    ##
    ## @return RTC::ReturnCode_t
    ##
    ##
    #def onRateChanged(self, ec_id):
    #
    #    return RTC.RTC_OK
	
def fooInit(manager):
    profile = OpenRTM_aist.Properties(defaults_str=foo_spec)
    manager.registerFactory(profile,
                            foo,
                            OpenRTM_aist.Delete)
def MyModuleInit(manager):
    fooInit(manager)
    # create instance_name option for createComponent()
    instance_name = [i for i in sys.argv if "--instance_name=" in i]
    if instance_name:
        args = instance_name[0].replace("--", "?")
    else:
        args = ""
  
    # Create a component
    comp = manager.createComponent("foo" + args)
def main():
    # remove --instance_name= option
    argv = [i for i in sys.argv if not "--instance_name=" in i]
    # Initialize manager
    mgr = OpenRTM_aist.Manager.init(sys.argv)
    mgr.setModuleInitProc(MyModuleInit)
    mgr.activateManager()
    mgr.runManager()
if __name__ == "__main__":
    main()
